02-express-tutorial/app.js

```diff
@@ -1 +1,62 @@
-console.log('Express Tutorial')
+const express = require('express')
+const { products } = require("./data");
+const app = express()
+app.use(express.static("./public"))
+
+// const idToFind = parseInt(req.params.productID); 
+// const product = products.find((p) => p.id === idToFind);
+
+app.get('/api/v1/products/:productID', (req, res) => {
+    const idToFind = parseInt(req.params); 
+    const product = products.find((p) => p.id === idToFind);
+    if (!product) {
+        // res.json({message: "That product was not found."})
+         return res.status(404).json({message: "That product was not found."})
+      }
+    res.json(product);
+  })
+
+app.get('/api/v1/products', (req, res) => {
+    let fProducts = [...products]
+    // const idToFind = parseInt(req.params.productID); 
+    // const product = products.find((p) => p.id === idToFind);
+    if (!fProducts) {
+        // res.json({message: "That product was not found."})
+         return res.status(404).json({message: "That product was not found."})
+      }
+    res.json(fProducts);
+  })
+
+  app.get('/api/v1/query', (req, res) => {
+    // console.log(req.query)
+    const { search, limit  } = req.query
+    let filteredProducts = [...products] // Copy the products array
+    if (search) {
+        filteredProducts = filteredProducts.filter((product) => {
+          return product.name.startsWith(search)
+        })
+      }
+      
+      if (limit) {
+        filteredProducts = filteredProducts.slice(0, Number(limit))
+      }
+      
+    filteredProducts = filteredProducts.filter(product => product.price > 20.00)
```

Comment:
Very close! You did the filter correctly, but it has to be in an if statement like before in order to have custom values entered (or for it to be active at all).

Reply:
I have solve the issue , please check Whenever it is convenient for you. Thank you for your comment